gnomad_qc/v5/data_ingestion/federated_validity_checks.py

```diff
@@ -197,22 +259,19 @@ def main(args):
                 "Filtering to %d partitions and sex chromosomes.", test_n_partitions
             )
             test_ht = ht._filter_partitions(range(test_n_partitions))
-            sex_ht = hl.filter_intervals(
-                ht,
-                [hl.parse_locus_interval("chrX")] + [hl.parse_locus_interval("chrY")],
-            )
-            ht = test_ht.union(sex_ht)
 
-        # Read in parameters from config file.
-        with hl.hadoop_open(config_path, "r") as f:
-            config = json.load(f)
+            x_ht = hl.filter_intervals(
+                ht, [hl.parse_locus_interval("chrX")]
+            )._filter_partitions(range(test_n_partitions))
+
+            y_ht = hl.filter_intervals(
+                ht, [hl.parse_locus_interval("chrY")]
+            )._filter_partitions(range(test_n_partitions))
+
+            ht = test_ht.union(x_ht, y_ht)
 
         # Validate config file against schema.
-        try:
-            validate(instance=config, schema=schema)
-            logger.info("JSON is valid.")
-        except ValidationError as e:
-            raise ValueError(f"JSON validation error: %s, {e.message}")
+        config = validate_config(ht=ht, config_path=config_path, schema=schema)
```

Comment:
Didnt think of it last time but I would move this to the top of the try since it is required, no need to run any code if the validation fails.

Reply:
i think it's as high as it can been since it's also testing if the fields are in the ht

Reply:
Well couldnt it go above the interval filtering? That shouldnt impact the fields right? But thats a good point, how do you feel about splitting the function into two? One is validate_config that just runs the schema validation in the try except and returns the config and then another like validate_ht_fields that runs all the checks on the table fields

Reply:
Yup I can split into two